cpp/tests/reductions/reduction_tests.cpp

```diff
@@ -135,21 +134,39 @@ TYPED_TEST(MinMaxReductionTest, MinMax)
   std::vector<int> int_values({5, 0, -120, -111, 0, 64, 63, 99, 123, -16});
   std::vector<bool> host_bools({1, 1, 0, 1, 1, 1, 0, 1, 0, 1});
   std::vector<bool> all_null({0, 0, 0, 0, 0, 0, 0, 0, 0, 0});
-  std::vector<T> v = convert_values<T>(int_values);
+  std::vector<T> v       = convert_values<T>(int_values);
```

Comment:
I was missing this. The `int_values` contains negative values, which definitely underflow. Here they are converted into `T` which may be an unsigned type, the converted results are wrapped around to be new positive values. Yes, using these all positive values (after conversion) still produces the correct results, but logically this is bad. We should not expect to have correct tests from invalid input.

So please also change all the negative values input into positive ones. This place, and also other places. I understand that this is part of an old work but we should fix it ASAP.

Reply:
Or, we can leave these input as-is, but change the test types. Instead of `AllTypes`, just select some types like `int32_t` and `float`. It is not necessary to test all types.